src/main.ts

```diff
@@ -240,19 +257,50 @@ export default class NewZettel extends Plugin {
   async makeNote(
     path: string,
     title: string,
-    content: string,
+    fileLink: string,
     placeCursorAtStartOfContent: boolean,
-    openZettel = false
+    openZettel = false,
+    successCallback: () => void = () => {
+      return;
+    }
   ) {
     const app = this.app;
     let titleContent = null;
     if (title && title.length > 0) {
-      titleContent = "# " + title + "\n\n";
+      titleContent = "# " + title;
     } else {
       titleContent = "";
     }
-    const fullContent = titleContent + content;
-    const file = await this.app.vault.create(path, fullContent);
+
+    let file = null;
+    const backlinkRegex = /{{link}}/g;
+    const titleRegex = /{{title}}/g;
+    if (this.settings.templateFile && this.settings.templateFile.trim() != "") {
+      let template_content = "";
+      try {
+        template_content = await this.app.vault.adapter.read(this.settings.templateFile.trim());
+      } catch (err) {
+        new Notice(
+          `[LUHMAN] Couldn't read template file. Make sure the path and file are valid/correct. Current setting: ${this.settings.templateFile.trim()}`,
+          15000
+        );
+        return;
+      }
+
+      if (!titleRegex.test(template_content) || !backlinkRegex.test(template_content)) {
+        new Notice("[LUHMAN] Template Malformed. Missing {{title}} and/or {{link}} placeholder. Please add them to the template and try again...", 15000);
+        return;
+      }
+
+      const file_content = template_content.replace(titleRegex, titleContent).replace(backlinkRegex, fileLink);
```

Comment:
This will clober 41ca033 can you please unclober that for me if not I will try to get to this some time this week but I am not 100% I can do that as I have a lot of stuff this week with easter coming up and all.

Reply:
Sorry I Accepted that one before looking at this one I probably would have had @electblake unclober instead if I had seen this one first but whats done is done.

Reply:
Manual Merging on my end and moving / fixing stuff there.